### ISSUE TITLE
How can I format YAxis Label

### ISSUE THREAD

**Issue by TuyenNguyenVan** [Abhinandan-Kushwaha/react-native-gifted-charts#22]
Could I show `1000` as  `$ 1K` in yAxis. Thanks

**Comment by Abhinandan-Kushwaha** [comment]
Hi @TuyenNguyenVan  👋
I have modified the custom Y-axis text feature.
Now, we send a separate array named **yAxisLabelTexts** as a prop.

This feature is available in version `0.1.8` 🎉

Here's the same example with the new update -
```

import { BarChart } from "react-native-gifted-charts";
        
const App = () => {
    const barData = [
        {value: 250, label: 'M'},
        {value: 500, label: 'T', frontColor: '#177AD5'},
        {value: 745, label: 'W', frontColor: '#177AD5'},
        {value: 320, label: 'T'},
        {value: 600, label: 'F', frontColor: '#177AD5'},
        {value: 256, label: 'S'},
        {value: 300, label: 'S'},
    ];
    return (
        <View>
            <BarChart
                barWidth={22}
                noOfSections={3}
                yAxisLabelTexts={['$0', '$250', '$500', '$750']}
                barBorderRadius={4}
                frontColor="lightgray"
                data={barData}
                yAxisThickness={0}
                xAxisThickness={0}
            />
        </View>
    );
};
```

**Comment by Abhinandan-Kushwaha** [comment]
> Nice. Thank @Abhinandan-Kushwaha for quick support. Do you have a plan to do a combined chart? Maybe in next, I need show a line chart in bar chart. Anyway, give you a start

Hi @TuyenNguyenVan Combined Line and Bar chart is now supported in version `0.2.0` 🎉

Just add the prop `showLine` to the `<BarChart/>` component. The properties of the Line chart can be controlled using the `lineConfig` object which has the following keys -


```js
type lineConfigType = {
  curved?: Boolean,
  isAnimated?: Boolean,
  delay?: number,
  thickness?: number,
  color?: ColorValue,
  hideDataPoints?: Boolean,
  dataPointsShape?: String,
  dataPointsWidth?: number,
  dataPointsHeight?: number,
  dataPointsColor?: ColorValue,
  dataPointsRadius?: number,
  textColor?: ColorValue,
  textFontSize?: number,
  textShiftX?: number,
  textShiftY?: number,
  shiftY?: number,
};
```

**Comment by jagajastic** [comment]
@Abhinandan-Kushwaha I still can't grasp how you format the <code> yAxisLabelText </code> 1000 as $ 1K in yAxis

**Comment by mmestiyak** [comment]
@Abhinandan-Kushwaha  I have two data to show in Area Chart, and need to get **yAxisLabelText**, i can't **yAxisLabelTexts** here, could you please help, any ideas?

**Comment by Abhinandan-Kushwaha** [comment]
Hi @mmestiyak,
Currently, the library does not support separate Y-Axis for multi-data sets. You are supposed to give a single array of `yAxisLabelTexts`.

If you provide an image illustrating your use case, I can work on adding a feature that will allow separate yAxisLabelTexts for separate data sets.

**Comment by Linuhusainnk** [comment]
could we format yAxis data which is automatically generated by data that we provider for xAxis

**Comment by ak-spotter** [comment]
Yes @Linuhusainnk, we can format the y Axis data which is automatically generated. 
The library generates numbers from the values present inside the data objects. You can add prefix or/and suffix to the y axis labels using `yAxisLabelPrefix` and `yAxisLabelSuffix` props.

**Comment by mphill** [comment]
It would be great to have something like `formatYLabel(value: number) : string`

**Comment by Abhinandan-Kushwaha** [comment]
Great suggestion @mphill 👍
I will add this prop in the next release.

**Comment by Abhinandan-Kushwaha** [comment]
Hi @mphill 👋
I have added the prop `formatYLabel` as per your suggestion.
The user can now pass a callback function that takes the label generated by the library and modifies it. The modified label returned by the callback function will be finally rendered along the Y-axis.

The prototype of the `formatYLabel` callback function is-
```js
(label: string) => string;
```
It is available in version `1.3.13`

**Comment by mphill** [comment]
I just tested.  It’s perfect.  Great addition. Thanks.

**Comment by dapeng0318** [comment]
Is there such function on LineChart as formatting min, max and median of the data on Y-axis to nearest decimal value and format such as thousands (K), millions (M) etc

If my max data is 15600000, then it shall display on the yaxis as 16M, while the median about 8000000 displays as 8M etc etc

**Comment by Abhinandan-Kushwaha** [comment]
@dapeng0318 
You can use the `formatYLabel` prop as shown below.
```js
<LineChart
  data = {data}
  formatYLabel = {(label) => {
    const labelVal = Number(label);
    if(labelVal >=1000000) return label.substring(0,label.length-6) + 'M';
    if(labelVal >= 1000) return label.substring(0,label.length-3) + 'K';
    return label;
  }}
/>

**Comment by dapeng0318** [comment]
@Abhinandan-Kushwaha 
I have another concern, given that my graph component is reusable, how do I go about setting different values?Graph on Tab 1 has max value of 500, on Tab 2 possibly 15M and Tab 3 possibly 8M.Running on same method  as well?

**Comment by Abhinandan-Kushwaha** [comment]
@dapeng0318 
You can set the `maxValue` as a state variable and update its value on tab switch. You can do the same for other props as well.

Using different chart components for each tab is also an option.

**Comment by dapeng0318** [comment]
@Abhinandan-Kushwaha 
Works great, thank you

**Comment by Oyedele-Sg** [comment]
Hey @Abhinandan-Kushwaha , please, just very similar to fomatYLabel, can we also have formatXLabel? So that it gives us the ability to format how we want the label on the x-axis to be. Thank you

**Comment by Oyedele-Sg** [comment]
Also, how can i increase the spacing on each section in x-axis. Some of my label text is getting cut off. Thanks @Abhinandan-Kushwaha

**Comment by Abhinandan-Kushwaha** [comment]
Hi @Oyedele-Sg 👋

### 1. formatXLabel
x-labels can be easily formatted, as we pass them manually with the data object. The library provides * fomatYLabel* because y-labels are auto-generated and we might want to format the auto-generated labels.

Here's an example of formatting your x-labels without having to use *formatXLabel*
```ts
const data = [
  {value: 15, label: '15'},
  {value: 30, label: '30'},
  {value: 26, label: '26'},
  {value: 40, label: '40'},
];

const prefix = '~';
const suffix = 'th';

data.forEach(item => item.label = prefix + item.label + suffix); // formatting the x-label

return <BarChart data={data} />
```
The output is

<img width="242" alt="Screenshot 2024-01-06 at 9 23 14 PM" src="https://github.com/Abhinandan-Kushwaha/react-native-gifted-charts/assets/20596944/fabdc157-0a73-4e38-a37f-4489ad70d4ec">

### 2. spacing on each section in x-axis
Just pass the `spacing` prop.
Here's the same example with `spacing={45}`

```ts
return <BarChart data={data} spacing={45}/>
```
<img width="355" alt="Screenshot 2024-01-06 at 9 26 32 PM" src="https://github.com/Abhinandan-Kushwaha/react-native-gifted-charts/assets/20596944/3bb055ae-f8e4-4a72-84dd-eb20fdd44e20">

**Comment by Oyedele-Sg** [comment]
Thank you so much @Abhinandan-Kushwaha. I was able to get it all done correctly. Thanks once again

**Comment by nguyenthanhdat4326** [comment]
Can u bring fomatYLabel to LineChart and xAxisType also, thanks @Abhinandan-Kushwaha

**Comment by RonanMaguire** [comment]
Hi, is there a way to format longer strings in the yAxis? I could probably reduce fontSize but if there is a more efficient way of doing this it would be better. See screenshot below. I want to be able to have digits plus km as a suffix, but as the digits grow the km becomes hidden behind the graph.

<img width="287" alt="Screenshot 2024-03-11 at 12 50 59" src="https://github.com/Abhinandan-Kushwaha/react-native-gifted-charts/assets/162998746/0acb9f47-ebb9-48ec-8ca5-e966a092c2c8">

Can I push the graph to the right slightly to accomodate this?
Thanks

**Comment by Scalete** [comment]
Hello, yAxisLabelTexts does not work correctly for me, if I write the following values ​​yAxisLabelTexts={['0', '1k', '2k', '3k', '4k', '5k', '6k']} then it only accepts 0, Nan displays the remaining numbers.

**Comment by sareyMediaLabIng** [comment]
> Hello, yAxisLabelTexts does not work correctly for me, if I write the following values ​​yAxisLabelTexts={['0', '1k', '2k', '3k', '4k', '5k', '6k']} then it only accepts 0, Nan displays the remaining numbers.

@Abhinandan-Kushwaha  im experiencing the same issue

**Comment by Abhinandan-Kushwaha** [comment]
Hi @Scalete @sareyMediaLabIng 👋
Thanks for reporting this issue. It is fixed in versions `1.4.13` onwards. 🎉
Please use the latest version of the library.
